src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -42,10 +42,20 @@ public StorageOperationException(String message) {
             super(message);
         }
     }
+    
+    /**
+     * Signals that the storage file has been deleted.
+     */
+    public static class FileDeletedException extends StorageOperationException {
```

Comment:
- in general, new exceptions warrant a class of their own. 
- you can only add new exceptions as nested class if they are totally local to the class and don't need to be used outside.
- in this case since it is `static` it is still OK to make it nested; point to ponder: understand how `static` nested classes are accessed!